blog/tech-talk-window-resize-behavior.md

```diff
@@ -0,0 +1,290 @@
+
+<!--truncate-->
+
+## Narrowing down the root cause
+
+I did a lot of experimentation.
+
+First, I noticed that the issue occurred in Google Chrome, too:
+
+![Screenshot of Google Chrome also showing the resize issue](/assets/img/blog/tech-talk-window-resize-behavior/google-chrome-repro.png)
+
+This suggested that the issue was likely in Chromium, not in Electron.
+
+Additionally, the issue was not visible on macOS. That suggested that it was in Windows-specific source code.
+
+## The crucial lead
+
+I tried a lot of different command line flags and configuration options.
+
+I noticed that [`app.disableHardwareAcceleration()`](../docs/latest/api/app#appdisablehardwareacceleration) fixed the issue. Without hardware acceleration, the issue was gone.
+
+Here is some context: Chromium supports various different graphics APIs for showing pixels on screen (OpenGL, Vulkan, Metal, and more). On Windows, it uses different graphics APIs than on Windows or Linux. Even on Windows, Chromium can work with multiple different graphics backends.
+
+Which graphics backend Chromium uses depends on the user's hardware. For example, some graphics backends require the computer to have a GPU.
+
+I tried various graphics backends and noticed that the following flags fixed the issue:
+
+- `--use-angle=warp`
+- `--use-angle=vulkan`
+- `--use-gl=desktop`
+- `--use-gl=egl`
+- `--use-gl=osmesa`
+- `--use-gl=swiftshader`
+
+The following flags reproduced the issue:
+
+- `--use-angle=d3d11` (this is currently the default on Windows)
+- `--use-angle=gl` (falls back to Direct3D 11 on Windows, see `chrome://gpu/`)
+
+None of the working flags were good enough to be used as the default in Electron apps on Windows. They were either too slow or lacked good driver support.
+
+However, these workarounds pointed me into the right direction. They showed that the issue was in a code path that was only used with the ANGLE Direct3D 11 backend.
+
+[Direct3D](https://en.wikipedia.org/wiki/Direct3D) is a Windows API for hardware-accelerated graphics.
+
+[ANGLE](<https://en.wikipedia.org/wiki/ANGLE_(software)>) is a library that translates OpenGL calls into calls to the native graphics API of the given operating system, here Direct3D. ANGLE allows Chromium developers to write OpenGL calls on all platforms. ANGLE then translates them into Direct3D, Vulkan, or Metal API calls, depending on which graphics API is used.
+
+## Locating the relevant Chromium component
+
+Chromium references Direct3D in tens of thousands of places. It wasn't realistic to go through all of them.
+
+By chance, I stumbled across a few helpful debugging flags in the Chromium source code:
+
+- `--ui-show-paint-rects`
+- `--ui-show-property-changed-rects`
+- `--ui-show-surface-damage-rects`
+- `--ui-show-composited-layer-borders`
+- `--tint-composited-content`
+- `--tint-composited-content-modulate`
+- (And more)
```

Comment:
Is there a specific file we want to reference here for folks who like to click through and follow along?